src/scripts/main.js

```diff
@@ -1,3 +1,224 @@
+    {},
+  );
+});
+
+function rebuildList(data) {
+  tableBody.innerHTML = '';
+
+  for (const item of data) {
+    const row = document.createElement('tr');
+
+    for (const key in item) {
+      const rowData = document.createElement('td');
+
+      rowData.innerText = item[key];
+      row.appendChild(rowData);
+    }
+
+    tableBody.appendChild(row);
+  }
+}
+
+tableHeader.addEventListener('click', (e) => {
+  const title = e.target.closest('th');
+
+  if (!title) {
+    return;
+  }
+
+  const targetValue = title.innerText.toLowerCase();
+
+  if (!title.hasAttribute('data-order') || title.dataset.order === DESC) {
+    title.setAttribute('data-order', ASC);
+  } else {
+    title.setAttribute('data-order', DESC);
+  }
+
+  items.sort((a, b) => {
+    const order = title.dataset.order;
+
+    if (targetValue === 'salary') {
+      const aSalary = a[targetValue].slice(1).split(',').join('');
+      const bSalary = b[targetValue].slice(1).split(',').join('');
+
+      return order === ASC ? aSalary - bSalary : bSalary - aSalary;
+    }
+
+    if (targetValue === 'age') {
+      return order === ASC
+        ? a[targetValue] - b[targetValue]
+        : b[targetValue] - a[targetValue];
+    }
+
+    return order === ASC
+      ? a[targetValue].localeCompare(b[targetValue])
+      : b[targetValue].localeCompare(a[targetValue]);
+  });
+
+  rebuildList(items);
+});
+
+tableBody.addEventListener('click', (e) => {
+  const activeRow = e.target.closest('tr');
+
+  if (!activeRow) {
+    return;
+  }
+
+  itemCollection.forEach((item) => {
+    item.classList.remove('active');
+  });
+
+  activeRow.classList.add('active');
+});
+
+function addForm() {
+  const body = document.body;
+  const form = document.createElement('form');
+
+  form.classList.add('new-employee-form');
+
+  for (const title of HEADERS) {
+    const label = document.createElement('label');
+
+    label.innerText = `${title[0].toUpperCase() + title.slice(1)}:`;
+
+    if (title === 'office') {
+      const selectBox = document.createElement('select');
+
+      selectBox.name = title;
+      selectBox.setAttribute('data-qa', title);
+      selectBox.required = true;
+
+      OFFICES.forEach((office) => {
+        const option = document.createElement('option');
+
+        option.value = office;
+        option.innerText = office;
+        selectBox.appendChild(option);
+      });
+
+      label.appendChild(selectBox);
+      form.appendChild(label);
+      continue;
+    }
+
+    const input = document.createElement('input');
+
+    input.name = title;
+    input.required = true;
+
+    if (title === 'age' || title === 'salary') {
+      input.type = 'number';
+      input.min = 0;
+    } else {
+      input.type = 'text';
+    }
+
+    label.appendChild(input);
+    form.appendChild(label);
+  }
+
+  const button = document.createElement('button');
+
+  button.type = 'submit';
+  button.innerText = 'Save to table';
+
+  form.appendChild(button);
+  body.appendChild(form);
+}
```

Comment:
The form creation logic can be broken down into smaller functions. For example, creating a label and input can be a separate function, creating a select box can be another function, and so on. This will make the code easier to understand and maintain.